### PR TITLE
git-together: init at 0.1.0-alpha.26

### PR DESCRIPTION
**Pull request by sentientmonkey**
## Description of changes

This adds [git-together](https://github.com/kejadlen/git-together) as a new nix pkg. This is a git wrapper that allows for git committer attribution when pair programming.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by arthsmn** [comment]
Please use the [by-name category](https://github.com/NixOS/nixpkgs/tree/master/pkgs#category-hierarchy) and follow the [commit conventions](https://github.com/NixOS/nixpkgs/tree/master/pkgs#commit-conventions). Also use the corrected commit message as the name of the PR.

**Review by arthsmn — COMMENTED** [review]
Shouldn't this package have a `meta.mainProgram`?

File: pkgs/applications/version-management/git-together/default.nix
Comment (arthsmn) [inline pkgs/applications/version-management/git-together/default.nix:26]:
```suggestion
    description = "Better commit attribution while pairing without messing with your git workflow";
```

File: pkgs/applications/version-management/git-together/default.nix
Comment (arthsmn) [inline pkgs/applications/version-management/git-together/default.nix:1]:
```suggestion
{ lib
, stdenv
, fetchFromGitHub
, rustPlatform
, openssl
, Security }:
```
This is the format usually used in nixpkgs.

**Comment by sentientmonkey** [comment]
> Shouldn't this package have a `meta.mainProgram`?

I can add one! Sorry, first nixpkg so I'm learning a bit 😁

File: pkgs/applications/version-management/git-together/default.nix
Comment (sentientmonkey) [reply pkgs/applications/version-management/git-together/default.nix:1]: Will do! I've been using `nixfmt` on save. Is there a different nixfmt config I can use to match nixpkgs?

**Comment by sentientmonkey** [comment]
> Please use the [by-name category](https://github.com/NixOS/nixpkgs/tree/master/pkgs#category-hierarchy) and follow the [commit conventions](https://github.com/NixOS/nixpkgs/tree/master/pkgs#commit-conventions). Also use the corrected commit message as the name of the PR.

Updated MR & will rebase to fix commit message

File: pkgs/applications/version-management/git-together/default.nix
Comment (arthsmn) [inline pkgs/applications/version-management/git-together/default.nix:25]:
```suggestion
  nativeBuildInputs = [pkg-config];

  buildInputs = [openssl] ++ lib.optional stdenv.isDarwin Security;

  OPENSSL_NO_VENDOR = true;

  OPENSSL_LIB_DIR=lib.mkIf stdenv.isDarwin (lib.getLib openssl);
```
Don't forget to also add pkg-config to the top

Comment (sentientmonkey) [reply pkgs/applications/version-management/git-together/default.nix]: That did the trick! The `OPENSSL_LIB_DIR` line didn't work, but it build ok without out on mac & linux. I added the `OPENSSL_NO_VENDOR` late in my attempts at getting libgit to look in the right place.

**Comment by sentientmonkey** [comment]
@arthsmn I _think_ this is in a much better state. Let me know if there's anything else I'm missing! Thanks for your help.

**Comment by arthsmn** [comment]
> > Shouldn't this package have a `meta.mainProgram`?
> 
> I can add one! Sorry, first nixpkg so I'm learning a bit 😁

Don't worry! You'll get the hang of it lol.

File: pkgs/applications/version-management/git-together/default.nix
Comment (arthsmn) [reply pkgs/applications/version-management/git-together/default.nix]: Oh, nice!

**Comment by arthsmn** [comment]
> @arthsmn I _think_ this is in a much better state. Let me know if there's anything else I'm missing! Thanks for your help.

I think you forgot the by-name category part, changing it will make the change to all-packages.nix unneeded. Other than this this seems just fine!

**Comment by sentientmonkey** [comment]
> > @arthsmn I _think_ this is in a much better state. Let me know if there's anything else I'm missing! Thanks for your help.
> 
> I think you forgot the by-name category part, changing it will make the change to all-packages.nix unneeded. Other than this it seems just fine!

Ah, that makes sense - I moved over to that convention - there was a minor change for referring to `darwin.Security` over `Security`, but that seems to work fine. The `by-name` validation script passed for me.

**Comment by arthsmn** [comment]
> > > @arthsmn I _think_ this is in a much better state. Let me know if there's anything else I'm missing! Thanks for your help.
> > 
> > 
> > I think you forgot the by-name category part, changing it will make the change to all-packages.nix unneeded. Other than this it seems just fine!
> 
> Ah, that makes sense - I moved over to that convention - there was a minor change for referring to `darwin.Security` over `Security`, but that seems to work fine. The `by-name` validation script passed for me.

Thanks for your contribution! Hope to see you here again.

**Review by arthsmn — APPROVED** [review]

File: pkgs/applications/version-management/git-together/default.nix
Comment (arthsmn) [reply pkgs/applications/version-management/git-together/default.nix]: > Will do! I've been using `nixfmt` on save. Is there a different nixfmt config I can use to match nixpkgs?

I'm not sure, need to search about.

File: pkgs/by-name/gi/git-together/package.nix
Comment (arthsmn) [inline pkgs/by-name/gi/git-together/package.nix:30]:
```suggestion
    license = licenses.mit;
```
Sorry, forgot to mention this! As it is just one license this seems better.

**Comment by sentientmonkey** [comment]
@piegamesde / @Janik-Haag is there anything else you need from me in order to merge?

**Review by Janik-Haag — COMMENTED** [review]
some formatting nits, other then that comments it looks good to me

**Comment by sentientmonkey** [comment]
@Janik-Haag all fixed up and rebased back into two commits again. Let me know if there are any other updates to make!

**Review by Janik-Haag — COMMENTED** [review]
one more thing [add maintainer sentientmonkey](https://github.com/NixOS/nixpkgs/pull/280382/commits/456b2d8e67bb10925a842dea93a2f994955b44ae) should be called maintainers: add sentientmonkey

**Comment by sentientmonkey** [comment]
@Janik-Haag all fixed!

**Review by Janik-Haag — APPROVED** [review]
LGTM, just waiting for ci

**Pull request merged**
